### DISCUSSION TITLE
Error network format body_with_headformat is not supported by cudNN backend

### DISCUSSION THREAD

**Discussion by JohnDijkstra** [LeelaChessZero/lc0#1904] [Q&A] [answered]
When attempting to load a T1 network (e.g t1-512×15×8) into LC0 0.30.0 with latest cudaNN package, running on a GTX 1050ti GPU, I am getting this error:

Error network format body_with_headformat is not supported by cudNN backend.

Any idea how to fix this?

**Comment by mooskagh** [comment, marked as answer]
Hi JohnDijkstra,

The cudnn backend hasn't been updated for quite long time. For newer networks, use CUDA version rather than cudNN.

**Reply by Spiderouf** [reply]
network.format

**Comment by JohnDijkstra** [comment]
Great, thanks! I'll try that in stead.

John Dijkstra
________________________________

**Reply by artemmiyy** [reply]
Hey JohnDijkstra,

Were you able to fix the issue? If so, how did you use CUDA instead of cudNN? Please let me know.

Regards,
Artemii

**Reply by skorokithakis** [reply]
`./lc0 --backend=cuda-auto`

**Comment by mooskagh** [comment]
> Were you able to fix the issue? If so, how did you use CUDA instead of cudNN? Please let me know.

Hi Artemii,

If you are running Lc0 under Windows, just download the version that has `cuda` in the name from here https://github.com/LeelaChessZero/lc0/releases (in the "Assets" section), rather than `cudnn`.

**Reply by artemmiyy** [reply]
I appreciate the fast reply. I'm running Linux and still not sure about how to fix the error, I have both backends (cuDNN and CUDA) installed with two GTX 1080 ti GPUs. Do you happen to know how to fix the error on Linux?

**Comment by mooskagh** [comment]
> I appreciate the fast reply. I'm running Linux and still not sure about how to fix the error, I have both backends (cuDNN and CUDA) installed with two GTX 1080 ti GPUs. Do you happen to know how to fix the error on Linux?

First make sure that your binary is indeed compiled both with CUDA and CudNN backends. To do that, run `./lc0 --help` and look at the list of possible values for the `--backend` parameter. `cuda-auto` should be one of them.

Then, to select that backend, either run `lc0` with `--backend=cuda-auto` parameter, or set `Backend` UCI parameter to `cuda-auto`, or create file `~/.config/lc0/lc0.config` and write `backend=cuda-auto` there.

For futher questions, I'd recommend to ask in the #help channel of our Discord chat at http://lc0.org/chat

**Reply by artemmiyy** [reply]
That worked – thank you!
